File: packages/react-native/temp/package/ReactAndroid/src/main/java/com/facebook/react/bridge/DimensionPropConverter.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.bridge;

import androidx.annotation.Nullable;
import com.facebook.yoga.YogaUnit;
import com.facebook.yoga.YogaValue;

public class DimensionPropConverter {

  @Nullable
  public static YogaValue getDimension(@Nullable Object value) {
    if (value == null) {
      return null;
    }

    if (value instanceof Double) {
      return new YogaValue(((Double) value).floatValue(), YogaUnit.POINT);
    }

    if (value instanceof String) {
      return YogaValue.parse((String) value);
    }

    throw new JSApplicationCausedNativeException(
        "DimensionValue: the value must be a number or string.");
  }
}

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/DimensionPropConverter.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.bridge

import com.facebook.yoga.YogaUnit
import com.facebook.yoga.YogaValue

internal class DimensionPropConverter {
  companion object {
    @JvmStatic
    fun getDimension(value: Any?): YogaValue? {
      return when (value) {
        null -> null
        is Double -> YogaValue(value.toFloat(), YogaUnit.POINT)
        is String -> YogaValue.parse(value)
        else ->
            throw JSApplicationCausedNativeException(
                "DimensionValue: the value must be a number or string.")
      }
    }
  }
}
